### ISSUE TITLE
Install request: CP2K 5.1

### ISSUE THREAD

**Issue by balston** [UCL-RITS/rcps-buildscripts#133]
IN02717168

Could you please compile the latest version of CP2K?

`svn checkout http://svn.code.sf.net/p/cp2k/code/branches/cp2k-5_1-branch cp2k`

I am running TDDFT calculations and they run much faster with the newer
version.

**Comment by heatherkellyucl** [comment]
Starting on this, first need to see whether our existing buildscript works or if there are changes needed to the arch file.

**Comment by heatherkellyucl** [comment]
Hmm, svn checkout on Legion node-y00a-034 keeps stalling partway through the src directory. Worked on login05 (and Thomas).

**Comment by heatherkellyucl** [comment]
Current arch file looks ok, but it does define `DATA_DIR   = /shared/ucl/apps/cp2k/4_1/gnu-4.9.2/data` which will need to change.

**Comment by heatherkellyucl** [comment]
The tests are going to take a while, it is now at 12 of 217.

**Comment by heatherkellyucl** [comment]
There's a downside with this install script for Grace/Thomas, which is that you don't really want to run the regression tests on a login node, but you `make test` them from the build directory. It uses 2 cores for hours.

Going to do a build with the regression tests in a job for those, before doing the central build.

**Comment by heatherkellyucl** [comment]
Bah. `svn: E000104: Error retrieving REPORT: Connection reset by peer` in job. Need an existing archive.

**Comment by heatherkellyucl** [comment]
In the meantime, one test on Legion failed, because of using the old data dir (I had since created a new 5_1 patches directory in buildscripts for the arch file with the new data path in it).

```
 *******************************************************************************
 *   ___                                                                       *
 *  /   \                                                                      *
 * [ABORT]   The specified OLD file <BASIS_LRIGPW_AUXMOLOPT> cannot be opened. *
 *  \___/                 It does not exist. Data directory path:              *
 *    |                  /shared/ucl/apps/cp2k/4_1/gnu-4.9.2/data              *
 *  O/|                                                                        *
 * /| |                                                                        *
 * / \                                                   common/cp_files.F:399 *
 *******************************************************************************


 ===== Routine Calling Stack ===== 

            3 read_qs_kind
            2 create_qs_kind_set   
            1 CP2K

...

--------------------------------- Summary --------------------------------
Number of FAILED  tests 1
Number of WRONG   tests 0
Number of CORRECT tests 2983
Number of NEW     tests 19
Total number of   tests 3003
GREPME 1 0 2983 19 3003 X
--------------------------------------------------------------------------
Regtest took 58831.00 seconds.
--------------------------------------------------------------------------
Thu Jan 11 05:49:52 GMT 2018
```
58831s is just over 16 hours...

**Comment by heatherkellyucl** [comment]
Doing a proper build on Legion, and jobs to run the regtests on Grace and Thomas, now that they are using local archives of the code.

**Comment by heatherkellyucl** [comment]
(We really could do with a build node on Grace and Thomas for this kind of thing)

I am thinking I should try this with OpenMPI 3.0.0 as well, as Owain built the current CP2K with 1.8.4 as 1.10.1 caused some tests to fail, and we have recently had some new versions of OpenMPI installed.

Can try this with a long qrsh session on Thomas at least.

**Comment by heatherkellyucl** [comment]
Legion OpenMPI 1.8.4 tests have succeeded!

```
--------------------------------- Summary --------------------------------
Number of FAILED  tests 0
Number of WRONG   tests 0
Number of CORRECT tests 2984
Number of NEW     tests 19
Total number of   tests 3003
GREPME 0 0 2984 19 3003 X
--------------------------------------------------------------------------
Regtest took 56274.00 seconds.
--------------------------------------------------------------------------
Fri Jan 12 04:06:18 GMT 2018
```

**Comment by heatherkellyucl** [comment]
Other OpenMPI 1.8.4 regtests.

Thomas (7.2hrs):
```
--------------------------------- Summary --------------------------------
Number of FAILED  tests 0
Number of WRONG   tests 0
Number of CORRECT tests 2984
Number of NEW     tests 19
Total number of   tests 3003
GREPME 0 0 2984 19 3003 X
--------------------------------------------------------------------------
Regtest took 25976.00 seconds.
--------------------------------------------------------------------------
Thu Jan 11 20:57:30 GMT 2018
```

Grace (5.2hrs):
```
--------------------------------- Summary --------------------------------
Number of FAILED  tests 0
Number of WRONG   tests 0
Number of CORRECT tests 2984
Number of NEW     tests 19
Total number of   tests 3003
GREPME 0 0 2984 19 3003 X
--------------------------------------------------------------------------
Regtest took 18654.00 seconds.
--------------------------------------------------------------------------
```
Test with `mpi/openmpi/3.0.0/gnu-4.9.2` on Thomas has begun - my qrsh session was naturally scheduled at 6:16pm so I've lost a large chunk of it...

**Comment by heatherkellyucl** [comment]
The tests with OpenMPI 3.0.0 are looking promising so far on Thomas so I've started a build with them on Legion and it'll be the central install if it works.

**Comment by heatherkellyucl** [comment]
Also running the tests as a job on Grace. If all works, should be ready for testless central builds on Thomas and Grace next week.

**Comment by heatherkellyucl** [comment]
Thomas with OpenMPI 3.0.0 (5.7hrs)
```
--------------------------------- Summary --------------------------------
Number of FAILED  tests 0
Number of WRONG   tests 0
Number of CORRECT tests 2993
Number of NEW     tests 19
Total number of   tests 3012
GREPME 0 0 2993 19 3012 X
--------------------------------------------------------------------------
Regtest took 20437.00 seconds.
--------------------------------------------------------------------------
Fri Jan 12 15:30:39 GMT 2018
```

**Comment by heatherkellyucl** [comment]
Testing done
- [ ] Legion
- [ ] Grace
- [ ] Thomas

Final installs done
- [ ] Legion
- [ ] Grace
- [ ] Thomas

**Comment by heatherkellyucl** [comment]
Module is available as `cp2k/5.1/ompi/gnu-4.9.2` on all clusters. Emailing requestor.

**Comment by heatherkellyucl** [comment]
Closing as no reports of problems.
